nushellPlugins.dbus: init at 0.10.0

DarkKronicle opened the pull request:
## Description of changes

Add [nu_plugin_dbus](https://github.com/devyn/nu_plugin_dbus)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

DarkKronicle committed:
maintainers: add darkkronicle

tobim reviewed (approved):
Hashes are correct, builds fine, and code follows the same patterns as the other plugins.

DarkKronicle commented:
I'm not sure if this will build on darwin (from what I can tell dbus should work on Mac?), all other plugins had some sort of added dependency. Does anyone know about that? I can wait a few more days for OfBorg to finish.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4433

SuperSandro2000 commented on pkgs/shells/nushell/plugins/default.nix:
```diff
@@ -1,10 +1,21 @@
+  dbus,
+}:
+lib.makeScope newScope (
+  self: with self; {
```
```suggestion
  self: {
```
this probably avoids the infinite recursion

DarkKronicle replied:
That worked, had to add `callPackage` as well.

SuperSandro2000 commented on pkgs/shells/nushell/plugins/dbus.nix:
```diff
@@ -0,0 +1,39 @@
+
+  cargoHash = "sha256-XmERSV2ihl2eS3PcT9hrP0PpTKiRic0Yz2UQyN+fqU8=";
+
+  nativeBuildInputs = [ pkg-config ] ++ lib.optionals stdenv.cc.isClang [ rustPlatform.bindgenHook ];
```
```suggestion
  nativeBuildInputs = [ pkg-config ] ++ lib.optionals stdenv.cc.isClang [ rustPlatform.bindgenHook ];

```

SuperSandro2000 commented on pkgs/shells/nushell/plugins/dbus.nix:
```diff
@@ -0,0 +1,39 @@
+  gitUpdater,
+  pkg-config,
+  dbus,
+}:
+let
+  version = "0.9.0";
+in
+rustPlatform.buildRustPackage {
+  inherit version;
+  pname = "nu-plugin-dbus";
```
```suggestion
}:

rustPlatform.buildRustPackage rec {
  pname = "nu-plugin-dbus";
  version = "0.9.0";
```

DarkKronicle committed:
nushellPlugins.dbus: init at 0.10.0

DarkKronicle commented:
Rebased and updated to use the newer nushell version. Added IOKit for Darwin because it failed to build, not sure if it needs CoreFoundation or Foundation.

DarkKronicle commented:
Closed because #349887
